### PR TITLE
=tes #24270 expectNoMessage should pull the unexpected msg 

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/24270

Nice to have in next release I guess.

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by ktoso** [comment]
Had a silly mistake in the test, will merge once validates

File: akka-testkit/src/main/scala/akka/testkit/TestKit.scala [hunk 1]
```diff
@@ -679,10 +679,18 @@ trait TestKitBase {
...
       }
     }
-    val diff = (max.toNanos - left.toNanos).nanos
-    val m = s"received unexpected message $elem after ${diff.toMillis} millis"
-    assert(elem eq null, m)
-    lastWasNoMsg = true
+
+    if (elem ne null) {
+      // we pop the message, such that subsequent expectNoMessage calls can
+      // assert on the nedxt period without a message
```
Comment (2m) [inline akka-testkit/src/main/scala/akka/testkit/TestKit.scala]: next

Comment (ktoso) [reply akka-testkit/src/main/scala/akka/testkit/TestKit.scala]: Nice catch!

**Comment by ktoso** [comment]
Squashing then

**Pull request merged**